src/RequestProcessor.cs

```diff
@@ -161,6 +201,17 @@ internal StreamingMessage ProcessFunctionLoadRequest(StreamingMessage request)
             }
             catch (Exception e)
             {
+                // If an exception is thrown while installing the function app dependencies,
+                // set response.FunctionLoadResponse.IsDependencyDownloaded to false.
+                if (e is DependencyInstallationException)
+                {
+                    response.FunctionLoadResponse.IsDependencyDownloaded = false;
+                }
+
+                // If a exception takes place during PowerShell initialization or while installing
+                // the function app dependencies, cache it so we can reuse it in future calls.
+                _unrecoverableFunctionLoadException = e;
```

Comment:
You are actually thread the exception thrown from `LoadFunction` as unrecoverable as well, which is wrong.

Reply:
This is fixed by your commit.